diff --git a/.gitignore b/.gitignore
new file mode 100644
index 0000000..0d20b64
--- /dev/null
+++ b/.gitignore
@@ -0,0 +1 @@
+*.pyc
diff --git a/rplugin/python3/denite/source/gitstatus.py b/rplugin/python3/denite/source/gitstatus.py
index 992d281..bb7e9f4 100644
--- a/rplugin/python3/denite/source/gitstatus.py
+++ b/rplugin/python3/denite/source/gitstatus.py
@@ -49,7 +49,7 @@ def run_command(commands, cwd):
     try:
         p = subprocess.run(commands,
                            cwd=cwd,
-                           encoding="utf8",
+                           universal_newlines=True,
                            stdout=subprocess.PIPE,
                            stderr=subprocess.STDOUT)
     except subprocess.CalledProcessError:
