diff --git a/autoload/eskk/dictionary.vim b/autoload/eskk/dictionary.vim
index 1903deb..9bbc609 100644
--- a/autoload/eskk/dictionary.vim
+++ b/autoload/eskk/dictionary.vim
@@ -1270,8 +1270,9 @@ function! s:ServerDict_request(command, key) dict "{{{
         if self.encoding != ''
             let key = iconv(key, &encoding, self.encoding)
         endif
-        call self._socket.write(printf("%s%s%s\n",
-        \ a:command, key, (key[strlen(key)-1] != ' ' ? ' ' : '')))
+        call self._socket.write(printf('%s%s%s%s',
+        \ a:command, key, (key[strlen(key)-1] != ' ' ? ' ' : ''),
+        \ self.last_cr ? "\n" : ''))
         let result = self._socket.read_line(-1, self.timeout)
         if self.encoding != ''
             let result = iconv(result, self.encoding, &encoding)
@@ -1279,7 +1280,7 @@ function! s:ServerDict_request(command, key) dict "{{{
 
         if result == ''
             " Reset.
-            call self._socket.write("0\n")
+            call self._socket.write(printf('0%s', self.last_cr ? "\n" : ''))
             call self._socket.close()
             call self.init()
         endif
@@ -1309,6 +1310,7 @@ let s:ServerDict = {
 \   'encoding': 'euc-jp',
 \   'timeout': 1000,
 \   'type': 'dictionary',
+\   'last_cr': 1,
 \
 \   'init': eskk#util#get_local_funcref('ServerDict_init', s:SID_PREFIX),
 \   'request': eskk#util#get_local_funcref('ServerDict_request', s:SID_PREFIX),
